### PR TITLE
[EventDispatcher] Show partial matching for debug:event-dispatcher

### LINKED ISSUE #14657: Search for pattern on debug:event-dispatcher
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#39323
| PR author(s) | @Nyholm
| Merged in    | 5.3-dev

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #14657

#SymfonyHackday

### CODE REVIEW

File: event_dispatcher.rst [hunk 1]
```diff
@@ -322,6 +322,17 @@ its name:
 
     $ php bin/console debug:event-dispatcher kernel.exception
 
+.. versionadded:: 5.3
+
+    You can also find listeners to event names partially matching a string. Example
```
Comment (OskarStark) [inline event_dispatcher.rst:327]:
```suggestion
    You can also find listeners to event names partially matching a string.
```

File: event_dispatcher.rst [hunk 2]
```diff
@@ -322,6 +322,17 @@ its name:
...
+    to show all events and their listeners related to ``kernel``, your app or Security:
+
+    .. code-block:: terminal
+
+        $ php bin/console debug:event-dispatcher kernel
+        $ php bin/console debug:event-dispatcher App
+        $ php bin/console debug:event-dispatcher Security
+
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion

to get everything matching "kernel" (i.e. ``kernel.exception``, ``kernel.response`` etc.) use:

.. code-block:: terminal

    $ php bin/console debug:event-dispatcher kernel

.. versionadded:: 5.3

    The ability to match partial event names was introduced in Symfony 5.3.
```

The advantage is, that we can just remove the versionadded in Symfony 6 docs, without rearranging the code examples itself.

Comment (Nyholm) [reply event_dispatcher.rst]: I see. Thank you

Comment (OskarStark) [reply event_dispatcher.rst]: Thank you, shall we add examples for App and or Security? Are these event names or does it matches a namespace?

File: event_dispatcher.rst [hunk 3]
```diff
@@ -322,6 +322,18 @@ its name:
 
     $ php bin/console debug:event-dispatcher kernel.exception
 
+
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
```

Comment (Nyholm) [reply event_dispatcher.rst]: oops =)

File: event_dispatcher.rst [hunk 4]
```diff
@@ -322,6 +322,17 @@ its name:
 
     $ php bin/console debug:event-dispatcher kernel.exception
 
+To get everything matching "kernel" (i.e. ``kernel.exception``, ``kernel.response`` etc.) use:
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
or can get everything matching "kernel" (i.e. ``kernel.exception``, ``kernel.response`` etc.):
```

Sorry for so many suggestions 🙈 💟

Comment (Nyholm) [reply event_dispatcher.rst]: Hmm. okey, you want to to match with the paragraph above.

Comment (OskarStark) [reply event_dispatcher.rst]: Yes it reads more fluently

File: event_dispatcher.rst [hunk 5]
```diff
@@ -322,6 +322,17 @@ its name:
 
     $ php bin/console debug:event-dispatcher kernel.exception
 
+or can get everything matching "kernel" (i.e. ``kernel.exception``, ``kernel.response`` etc.):
+
+.. code-block:: terminal
+
+    $ php bin/console debug:event-dispatcher kernel
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
    $ php bin/console debug:event-dispatcher kernel // matches "kernel.exception", "kernel.response" etc.
```

File: event_dispatcher.rst [hunk 6]
```diff
@@ -322,6 +322,17 @@ its name:
 
     $ php bin/console debug:event-dispatcher kernel.exception
 
+or can get everything matching "kernel" (i.e. ``kernel.exception``, ``kernel.response`` etc.):
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
or can get everything which partial matches the event name:
```

File: event_dispatcher.rst [hunk 7]
```diff
@@ -322,6 +322,17 @@ its name:
...
+
+.. code-block:: terminal
+
+    $ php bin/console debug:event-dispatcher kernel
+    $ php bin/console debug:event-dispatcher App
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
    $ php bin/console debug:event-dispatcher Security // matches "Symfony\Component\Security\Http\Event\CheckPassportEvent"
```

I took this from your code PR

**Review by OskarStark — COMMENTED** [review]
What do you think @Nyholm

**Comment by Nyholm** [comment]
Excellent. Thank you

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thanks Tobias for working on this feature, this is much appreciated.

**Comment by Nyholm** [comment]
Thank you for the help and for merging

**Pull request merged**